tools/istio-iptables/pkg/cmd/root.go

```diff
@@ -136,6 +136,8 @@ func bindCmdlineFlags(cfg *config.Config, cmd *cobra.Command) {
 		&cfg.NetworkNamespace)
 
 	flag.BindEnv(fs, constants.CNIMode, "", "Whether to run as CNI plugin.", &cfg.CNIMode)
+
+	flag.BindEnv(fs, constants.PreemptiveCleanup, "", "Whether to run a preemptive cleanup to prevent duplicated rules and chains", &cfg.PreemptiveCleanup)
```

Comment:
Asking, because I haven't formed an opinion yet - does this need to be optional? Why would you not want to always just do this as a pre-step?

Reply:
I was not sure if it should be performed all the time. I put it as opt-in for now in order to not impact the VM usecase (usecase of which I don't know much about).
I'm investigating if there could be an impact there by doing a preemptive cleanup which deletes the ISTIO_* chains and every rule in them.

Anyway, I'm still doing some manual tests and I need to add proper coverage for this cleanup.

Reply:
I would prefer _not_ to make this a flag, and simply do it every time, which should be fine if it is idempotent.

If we did have to make it an option, the option should be an opt-out, not an opt-in.